src/Button/ButtonBase.tsx

```diff
@@ -26,8 +30,15 @@ const ButtonBase = forwardRef<HTMLElement, ButtonProps>(
       return merge(baseStyles, sxProp as SxProp)
     }, [baseStyles, sxProp])
 
+    React.useEffect(() => {
+      if (!(innerRef.current instanceof HTMLButtonElement) && !(innerRef.current instanceof HTMLAnchorElement)) {
+        // eslint-disable-next-line no-console
+        console.warn('This component should be an instanceof a semantic button or anchor')
```

Comment:
I think one think we'll need to add is an `if (__DEV__)` check to this so that it only is emitted in development. 

```suggestion
        if (__DEV__) {
          console.warn('This component should be an instanceof a semantic button or anchor')
        }
```

Hope that makes sense!